### DISCUSSION TITLE
Error migrating Kong OSS from 2.8.3 to 3.1.1.

### DISCUSSION THREAD

**Discussion by mayankmurari** [Kong/kong#11421] [Help] [answered]
I am trying to migrate Kong from 2.8.3 to 3.1.1. Kong 3.1.1 pods are not coming up and below error is present in the logs.
Current Kong version is 2.8.3 with KIC 2.8.2 and Postgres DB.

```
 [emerg] 1#0: bind() to unix:/kong_prefix/worker_events.sock failed (98: Address in use)
 [notice] 1#0: try again to bind() after 500ms```

**Comment by chronolaw** [comment, marked as answer]
Kong 3.x introduced a new events library called `lua-resty-events`, the blog is [here](https://konghq.com/blog/engineering/nginx-openresty-event-handling-strategy-for-cpu-efficiency).

Since it is a new library there are some bugs about it. I suggest you upgrade to the latest kong, like Kong 3.4, which may fix this issue.

**Reply by mayankmurari** [reply]
@chronolaw Thanks for your input.
